### DISCUSSION TITLE
Middleware causing post requests larger than 64kb to hang

### DISCUSSION THREAD

**Discussion by arvind-agarwal** [vercel/next.js#39592] [Help]
I am facing an issue with large post body, If the post body size is larger than 64k it hangs the client for a long time and is not processed. I zeroed it down on issue with middleware. If I introduce the middleware (even that does not do anything), than the post request does not succeed.

Steps to reproduce.
1. Create a brand new project using create-next-app
2. Create a middleware.js file in root of the project
```
export function middleware(request) {
  console.log(request.url);
}
```
3. Start the above server using 
```
npm run dev
```
4. Start another terminal session
5. Install axios
```
npm install axios 
```
6. Create a file in root testRequest.js, needs axios as dependency
```
const axios=require("axios");

axios.post("http://localhost:3000/test", {
  "x": '1'.repeat(1024*64)
}).then(function (response) {
  console.log(response);
})
.catch(function (error) {
  console.log(error);
});
```
7. Run the file using node testRequest.js. This process will hang and will not finish.

If I reduce the request size by changing multiple in repeat to 63, the request will not hang and some response will be returned.

Any help in what maybe causing this issue or possibly workaround for the same will be appreciated.

Thanks

**Comment by braden-catlett** [comment]
I'm experiencing this as well. You have to use the middleware config matcher property to limit the middleware from running on API routes.
